### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by AlexJanow**

### CODE REVIEW

**Review by mpagels — CHANGES_REQUESTED** [review]
Without doing anything, just visiting the site, it gives me this:
Bildschirmfoto 2021-07-22 um 01 39 08
It seems that something is `fetching` on mount of the component. Maybe you can fix this, before merging the pull request into production.

But overall, I see a lot of good things here. I left some comments. Please have a look.

Very good job, Alex.

File: src/App.css [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+.App__nav a.active {
+  font-weight: bold;
+  text-decoration: whitesmoke underline;
+}
+.App__nav a {
+  text-decoration: none;
+  color: black;
+}
+
+.uber__div {
```
Comment (namirsab) [inline src/App.css:35]: Classnames shouldn't have the name of the tag. That's just a general CSS tip

File: src/App.css [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+}
+.App__footer {
+  padding: 1rem;
+  border-top: 1px solid black;
+  position: relative;
+  margin-top: auto;
+  bottom: 0;
+  height: auto;
+  font-size: 2rem;
+}
```
Comment (namirsab) [inline src/App.css:20]: I went to the app and tried to play around with many of these properties, and I have the feeling they are not doing much. Just go to the inspector and try to remove a few of the, and you'll see they are not needed.

File: src/App.css [hunk 3]
```diff
@@ -1,3 +1,56 @@
 .App {
```
Comment (namirsab) [inline src/App.css:1]: Just a general thing.
The layout could be achieved much easier with a grid or a flex. I think you could take a  look how John structured his app using a grid.

Comment (namirsab) [reply src/App.css:1]: Also check the general comments I made to John, some of them apply to your code as well:
https://github.com/johnxgaehner/tell-me-your-name/pull/1/files#r674625090

File: src/pages/Age.css [hunk 4]
```diff
@@ -0,0 +1,36 @@
+main {
+  width: fit-content;
+  margin: auto;
+}
```
Comment (namirsab) [inline src/pages/Age.css:4]: You are styling everything again and again in every page. If you check your Age.css and your Gender.css they are almost the same.

Also, in CSS, once a selector is defined in one place it's global. If the same selector is defined more than one time, then the last read selector would take over

Comment (namirsab) [reply src/pages/Age.css:4]: Also, this is telling you that your some parts of your pages could be independent components, like the form for instance. You could then reuse those components

File: src/pages/Age.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    const url = `https://api.agify.io?name=${search}`;
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => {
+        setName(data.name);
+        setAge(data.age);
+
+        setIsLoading(false);
+      });
+  }, [search]);
```
Comment (namirsab) [inline src/pages/Age.js:30]: This useEffect is also running on mount. Maybe you want to get rid of the useEffect (and just fetch in the handleSubmit or at least guard the useEffect callback function so it only does the fetch if the name state is defined/truthy

File: src/pages/Age.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  return (
+    <div>
+      <header>
+        <h1>AGE</h1>
+      </header>
```
Comment (namirsab) [inline src/pages/Age.js:36]: The header should be outside the main pages. You could take a look how john solved it for instance.

**Review by namirsab — COMMENTED** [review]
Good Job Alex!
Clearly you are getting the grasp of react. Next step is gettign the grasp of creating individual components and reusing them in different places.

I also know that for the time you had you did an amazing job!

I left some comments, please take a loook :)

File: src/App.js [hunk 7]
```diff
@@ -1,7 +1,48 @@
...
+              <Nation />
+            </Route>
+            <Route path="/">
+              <Age />
+            </Route>
+          </Switch>
+        </main>
+        <footer className="App__footer">
+          <nav className="App__nav">
+            <NavLink exact to="/">
```
[the remark this answers is not in the document]
Comment (AlexJanow) [reply src/App.js:31]: If I remove it "Age" gets bold ( made css bold on activated link) :/

**Pull request closed without merging**